Network/DNS/Transport.hs

```diff
@@ -23,8 +23,17 @@ import Network.DNS.Types.Internal
 -- pipelined TCP, we'll need to handle out of order responses.  See:
 -- https://tools.ietf.org/html/rfc7766#section-7
 checkResp :: Question -> Identifier -> DNSMessage -> Bool
-checkResp q seqno resp =
-   identifier (header resp) == seqno && [q] == question resp
+checkResp q seqno = isNothing . checkRespM (Just q) seqno
```

Comment:
There's no reason to complicate this interface. Some duplication of code is harmless in the name of simplicity. The function that does not check the question can repeat the code to check the transaction id.